test/test_source/pp/preproc_elif.F90

```diff
@@ -0,0 +1,139 @@
+subroutine preprocessor_elif(var, var3, var4, var5, var6)
+
+#if 0
+#define MYTYPE character
+integer, intent(in) :: var
+#elif 1
+#define MYTYPE logical
+integer, intent(inout) :: var
+var = 3
+#else
+#define MYTYPE integer
+integer, intent(out) :: var
+var = 5
+#endif
+
+MYTYPE :: var1
+
+#define OTHERTYPE integer
+
+OTHERTYPE :: var2
+
+PRINT*, var
+
```

Comment:
This is (amongst others) a regression test, to ensure that a bug that was present, does not re-occur.

It checks that the steps after the preprocessor parsing has fully finished, are only using content from the parts within the preprocessor if-elif-else that should be used. To do this, it has some regular fortran code within the #if and #elif.
And, it checks that the #endif correctly concludes the if-elif, so any new #define statements that come after the #endif, are picked up during the preprocessor parsing.

Maybe it can be reduced a bit, it may have some content that is not necessary to check this.